### PR TITLE
Gulpfile Edits

### PR DESCRIPTION
**Pull request by theblobscp**
Makes Gulpfile look better

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -19,34 +19,33 @@ const finalMinSCSSName = "devicon.min.scss";
...
-        createAliasSCSS(deviconJson),
-        createColorsCSS(deviconJson)
-    ])
+  const deviconJson = JSON.parse(
+    await fsPromise.readFile(path.join(__dirname, deviconJSONName), "utf8")
+  );
+
+  await Promise.all([
+    createAliasSCSS(deviconJson),
+    createColorsCSS(deviconJson),
```
Comment (Thomas-Boi) [inline gulpfile.js:47]:
```suggestion
    createColorsCSS(deviconJson)
```

File: gulpfile.js [hunk 2]
```diff
@@ -55,123 +54,119 @@ async function createCSSFiles() {
...
-    let {
-        name,
-        aliases
-    } = fontObj;
+  let { name, aliases } = fontObj;
 
-    return aliases.map(aliasObj => {
-        return `.devicon-${name}-${aliasObj.alias} {
+  return aliases
+    .map((aliasObj) => {
```
Comment (Thomas-Boi) [inline gulpfile.js:80]:
```suggestion
    .map(aliasObj => {
```

File: gulpfile.js [hunk 3]
```diff
@@ -55,123 +54,119 @@ async function createCSSFiles() {
...
-        });
-
-        return `${cssClasses.join(",")}{color: ${color}}`;
-    }).join(" ");
-
-    let cssPath = path.join(__dirname, colorsCSSName);
-    return fsPromise.writeFile(cssPath, statements, "utf8");
+  // create the color statements for each font object
+  let statements = deviconJson
+    .map((fontObj) => {
```
Comment (Thomas-Boi) [inline gulpfile.js:98]:
```suggestion
    .map(fontObj => {
```

File: gulpfile.js [hunk 4]
```diff
@@ -55,123 +54,119 @@ async function createCSSFiles() {
...
-                return fsPromise.unlink(filePath);
-            } catch(e) {
-                console.log(e);
-            }
-        })
-    );
+  let fileNames = [aliasSCSSName, colorsCSSName, finalMinSCSSName];
+
+  return Promise.all(
+    fileNames.map((name) => {
```
Comment (Thomas-Boi) [inline gulpfile.js:135]:
```suggestion
    fileNames.map(name => {
```

File: gulpfile.js [hunk 5]
```diff
@@ -181,20 +176,19 @@ function configOptionCallback(file) {
       {
         prefixIds: {
           prefix: file.stem, // add file name to ids
-          delim: "-"
-        } 
+          delim: "-",
```
Comment (Thomas-Boi) [inline gulpfile.js:179]:
```suggestion
          delim: "-"
```

File: gulpfile.js [hunk 6]
```diff
@@ -181,20 +176,19 @@ function configOptionCallback(file) {
...
+        },
       },
       {
-        removeViewBox: false // keep viewbox
+        removeViewBox: false, // keep viewbox
```
Comment (Thomas-Boi) [inline gulpfile.js:183]:
```suggestion
        removeViewBox: false // keep viewbox
```

File: gulpfile.js [hunk 7]
```diff
@@ -181,20 +176,19 @@ function configOptionCallback(file) {
...
       },
       {
-        removeDimensions: true // remove height and width
-      }
-    ]
-  }
+        removeDimensions: true, // remove height and width
```
Comment (Thomas-Boi) [inline gulpfile.js:186]:
```suggestion
        removeDimensions: true // remove height and width
```

File: gulpfile.js [hunk 8]
```diff
@@ -181,20 +176,19 @@ function configOptionCallback(file) {
...
+      },
```
Comment (Thomas-Boi) [inline gulpfile.js:187]:
```suggestion
      }
```

File: gulpfile.js [hunk 9]
```diff
@@ -181,20 +176,19 @@ function configOptionCallback(file) {
...
+    ],
```
Comment (Thomas-Boi) [inline gulpfile.js:188]:
```suggestion
    ]
```

File: gulpfile.js [hunk 10]
```diff
@@ -55,123 +54,119 @@ async function createCSSFiles() {
...
-    let cssPath = path.join(__dirname, colorsCSSName);
-    return fsPromise.writeFile(cssPath, statements, "utf8");
+  // create the color statements for each font object
+  let statements = deviconJson
+    .map((fontObj) => {
+      let {
+        name,
+        versions: { font: fonts },
+        color,
+        aliases,
```
Comment (Thomas-Boi) [inline gulpfile.js:103]:
```suggestion
        aliases
```

File: gulpfile.js [hunk 11]
```diff
@@ -55,123 +54,119 @@ async function createCSSFiles() {
...
+      if (fonts.length === 0 || typeof color !== "string") {
+        console.log(`This object doesn't have a font or a color: ${name}`);
+        return "";
+      }
+
+      // process the icons in the font attr
+      let cssClasses = fonts.map((font) => `.devicon-${name}-${font}.colored`);
+
+      // process the icons in the aliases attr
+      aliases.forEach((aliasObj) => {
```
Comment (Thomas-Boi) [inline gulpfile.js:115]:
```suggestion
      aliases.forEach(aliasObj => {
```

File: gulpfile.js [hunk 12]
```diff
@@ -55,123 +54,119 @@ async function createCSSFiles() {
...
+  let svgGlob = JSON.parse(yargs.argv.svgFiles);
+  console.log("Optimizing these files: ", svgGlob);
+  return gulp
+    .src(svgGlob)
     .pipe(svgmin(configOptionCallback))
-    .pipe(gulp.dest(file => {
-      return file.base
-    }))
+    .pipe(
+      gulp.dest((file) => {
```
Comment (Thomas-Boi) [inline gulpfile.js:161]:
```suggestion
      gulp.dest(file => {
```

**Review by Thomas-Boi — CHANGES_REQUESTED** [review]
Overall, the suggested changes are good and I appreciate them. However, there are a few that don't fit the repo's style, such arrow callback with one argument using parentheses or dangling commas in objects. 

If you can address those issues, the PR is good to go. Do take a look at the file to make sure everything is gone though, I might have miss some of them.

**Comment by Thomas-Boi** [comment]
For future references, please avoid making PRs that contains only stylistic changes that don't bring a lot of benefits to the repo. If you are refactoring an absolute mess of a file, that'll be greatly appreciated. However, for files like the `gulpfile` that don't get worked on often, this is unnecessary.

I know that many of the files are not 100% up to industry gold standards. However, they do meet many of these stylistic standards (readable code, properly commented) and they work. Furthermore, this is a hobby. I don't want to force everyone to follow every small stylistic rules, hence the lack of linters in the project. Instead, if you can dedicate that time to features, that'll be much more appreciated.

**Comment by theblobscp** [comment]
Alright. Although to be honest, that's kinda all I'm good at. But it is what it is

**Comment by Thomas-Boi** [comment]
> Alright. Although to be honest, that's kinda all I'm good at. But it is what it is

You can still help out with feature request and icon reviews. Those are easy to pick up and we can always use a hand with them.

Also, thanks for fixing the issues but there are a few left. Just a tip: you can fix them quickly on the website by clicking on "Add suggestion to batch" or "Commit Suggestion"
image

This will let GitHub commit my changes in the PR without the need of you relooking for the suggestions.

**Comment by theblobscp** [comment]
> Just a tip: you can fix them quickly on the website by clicking on "Add suggestion to batch" or "Commit Suggestion"

Thanks, I never knew that

**Review by Thomas-Boi — APPROVED** [review]

**Comment by Thomas-Boi** [comment]
Good work 👍. Everything looks good now. 

Remember, there are many ways to contribute. Just because we don't accept some of them doesn't mean that we don't accept everything. If you want, you can also open an Issue or message us on Discord before you want to do something. That way, you don't have to lose time on your PRs.

**Comment by theblobscp** [comment]
> you can also open an Issue or message us on Discord

Can I get the link

**Comment by Thomas-Boi** [comment]
@theblobscp here you go: https://github.com/devicons/devicon#discord-server

**Comment by Panquesito7** [comment]
> > you can also open an Issue or message us on Discord
> 
> Can I get the link

Here's the link invitation to our Discord server: https://discord.gg/hScy8KWACQ

**Comment by theblobscp** [comment]
Thanks to you both

**Pull request merged**